peer/peer_test.go

```diff
@@ -82,7 +85,9 @@ func TestPeerStringer(t *testing.T) {
 	}
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			ctx := NewContext(context.Background(), tc.peer)
+			ctx, cancel := context.WithTimeout(NewContext(context.Background(), tc.peer), defaultTestTimeout)
```

Comment:
Again, I think splitting in multiple statements would make this a bit more readable.

Reply:
Fixed.